datastore/src/test/java/io/spine/server/storage/datastore/DsAggregateStorageShould.java

```diff
@@ -98,6 +109,56 @@ public void fail_to_write_invalid_record() {
                             AggregateEventRecord.getDefaultInstance());
     }
 
+    @Test
+    public void set_limit_for_history_backward_query() {
+        final DsAggregateStorage<ProjectId> storage = (DsAggregateStorage<ProjectId>) getStorage();
+        final int batchSize = 10;
+        final AggregateReadRequest<ProjectId> request = new AggregateReadRequest<>(newId(),
+                                                                                   batchSize);
+        final EntityQuery historyBackwardQuery = storage.historyBackwardQuery(request);
+
+        final int queryLimit = historyBackwardQuery.getLimit();
+        assertEquals(batchSize, queryLimit);
+    }
+
+    @Test
+    public void still_load_aggregates_properly_after_snapshot_trigger_decrease_at_runtime() {
+        final ProjectAggregateRepository repository = new ProjectAggregateRepository();
+        repository.initStorage(datastoreFactory);
+        repository.setBoundedContext(BoundedContext.newBuilder()
+                                                   .build());
+        final ProjectId id = newId();
+        final int initialSnapshotTrigger = 10;
+
+        // To restore an aggregate using a snapshot and events.
+        final int tasksCount = initialSnapshotTrigger * 2 - 1;
+
+        repository.setSnapshotTrigger(initialSnapshotTrigger);
+        final TestActorRequestFactory factory = newInstance(DsAggregateStorageShould.class);
+        for (int i = 0; i < tasksCount; i++) {
+            final AggAddTask command = addTask(id);
+            final CommandEnvelope envelope = CommandEnvelope.of(factory.createCommand(command));
+            repository.dispatch(envelope);
+        }
+
+        final int minimalSnapshotTrigger = 1;
+        repository.setSnapshotTrigger(minimalSnapshotTrigger);
+        final ProjectAggregate aggregate = repository.find(id)
+                                                     .get();
+        assertEquals(tasksCount, aggregate.getState()
+                                          .getTaskCount());
+    }
+
+    private static class ProjectAggregateRepository
```

Comment:
Please move this class to `Given` (like we do in `core-java`) and have a class-level comment for it.